.github/workflows/github-actions-bazel-build.yml

```diff
@@ -0,0 +1,51 @@
+name: bazel-build
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+jobs:
+  Build:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Create Cache Timestamp
+        id: cache_timestamp
+        uses: nanzm/get-time-action@v2.0
+        with:
+          format: 'YYYY-MM-DD-HH-mm-ss'
+
+      - name: Mount bazel cache
+        uses: actions/cache@v4
+        with:
+          path: "~/.cache/bazel"
+          key: bazelcache_${{ steps.cache_timestamp.outputs.time }}
```

Comment:
If the key contains the timestamp how does it get hits?  Wouldn't the key always differ?

Reply:
The look-up looks for the prefix `bazelcache_` (see the `restore-keys` part), and then sorts by filename. So with this set-up we automatically get the latest cache that has been written.
The filename is _written_ with the timestamp.